### PR TITLE
Add training parameter to dropout to make it work

### PR DESCRIPTION
**Pull request by PiotrCzapla**
I think that without this parameter set dropout is disabled all the time. At least this is what I read in the documentation, besides adding this improves training.

### CODE REVIEW

**Comment by PiotrCzapla** [comment]
I signed the development agreement.

File: tensorflow/examples/learn/mnist.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def conv_model(features, labels, mode):
   # Densely connected layer with 1024 neurons.
   h_fc1 = tf.layers.dense(h_pool2_flat, 1024, activation=tf.nn.relu)
   if mode == tf.estimator.ModeKeys.TRAIN:
-    h_fc1 = tf.layers.dropout(h_fc1, rate=0.5)
+    h_fc1 = tf.layers.dropout(h_fc1, rate=0.5, training=mode == tf.estimator.ModeKeys.TRAIN)
```
Comment (qmick) [inline tensorflow/examples/learn/mnist.py]: It is equivalent to `training=True`

Comment (rmlarsen) [reply tensorflow/examples/learn/mnist.py]: Not if called from evaluation.

**Review by rmlarsen — APPROVED** [review]

File: tensorflow/examples/learn/mnist.py (hunk 1 above)
Comment (rmlarsen) [inline tensorflow/examples/learn/mnist.py]: Add parens around the logical expression.

**Review by rmlarsen — CHANGES_REQUESTED** [review]

File: tensorflow/examples/learn/mnist.py (hunk 1 above)
Comment (qmick) [reply tensorflow/examples/learn/mnist.py]: if called from evaluation this line won't be executed.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rmlarsen** [comment]
@PiotrCzapla any update?

**Comment by PiotrCzapla** [comment]
@rmlarsen, I've removed the redundant if statement around dropout, and added parenthesis as requested. Maybe we can squash this two commits?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Review by rmlarsen — APPROVED** [review]

**Comment by rmlarsen** [comment]
@PiotrCzapla please fix the linter errors: https://source.cloud.google.com/results/invocations/2b1cc07a-4044-4ef2-a26e-a5f31a80316d/log

**Review by qmick — APPROVED** [review]
